cmd/copy.go

```diff
@@ -1163,7 +1163,7 @@ func init() {
 	// options change how the transfers are performed
 	cpCmd.PersistentFlags().Float64Var(&raw.blockSizeMB, "block-size-mb", 0, "use this block size (specified in MiB) when uploading to/downloading from Azure Storage. Default is automatically calculated based on file size. Decimal fractions are allowed - e.g. 0.25")
 	cpCmd.PersistentFlags().StringVar(&raw.logVerbosity, "log-level", "INFO", "define the log verbosity for the log file, available levels: INFO(all requests/responses), WARNING(slow responses), ERROR(only failed requests), and NONE(no output logs).")
-	cpCmd.PersistentFlags().StringVar(&raw.blobType, "blob-type", "None", "defines the type of blob at the destination. This is used in case of upload / account to account copy")
+	cpCmd.PersistentFlags().StringVar(&raw.blobType, "blob-type", "None", "defines the type of blob at the destination. This is used in case of upload / account to account copy. Use --blob-type detect for auto-detection.")
```

Comment:
Fair point, though, read my comment on the above conversation. (Basically using detect as a way to explicitly define detection for s2s copies, and defaulting to the source type if not explicitly defined.)

Reply:
We can't say "Use detect for auto-detection" if auto detection also happens with None. That's misleading to users, since it implies that auto detection will not happen with anything other than "detect".

I suspect that the benefit of having None and Detect (which applies only to S2S from blob storage) is not enough to justify the complexity of having both and _explaining_ both to users.

Reply:
I think the 3 options can perhaps exist side by side, but with the following meanings:

- None: no detection at all, use block blob by default (for both upload and S2S).
    - For context, people may backup page blobs into block blobs to save on cost.
- Detect: smartly infer which file should be uploaded as page blob when uploading, and preserve the source type when S2S.
    - Detect may be necessary since the user may not want "smart" behavior by default, so we should let them opt-in.
- Specific blob type: override, upload all files to this kind of blobs
    - Basically telling AzCopy to shut up and just do the transfer hihi.

Reply:
I'm thinking that for File -> Blob and S3 -> Blob copies, if --blob-type=detect, we should infer the VHD files just like uploading. I'll confirm the expected behavior with the Stg Exp Team.

Please let me know your thoughts.

Reply:
Yeah, confirming with Stg Exp team sounds good. As does inference same is uploading for copying from File and S3.

Reply:
Re the 3 possibilities of None, Detect and giving a specific type, wouldn't "None" be identical to - - blob-type BlockBlob?  Is that confusing to us or to users? Or maybe it would be more confusing, in the code, if we did _not_have None. So maybe have both None and BlockBlob, but don't document None at all???

Reply:
I mean, None should stick around. None is intended for Files and ADLSG2.

I do think that it may be a good idea, usability wise, to hide None from the user and just default to BlockBlob; switching to None for Files and ADLSG2 transfers.